src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java

```diff
@@ -57,4 +63,116 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
         }
         return sb.toString();
     }
+
+    /**
+     * Will check the config file to see if players are allowed to use the command that is sent here.
+     * and will verify that they are of correct rank to do so.
+     * @param player the players/senders name.
+     * @param theCommand which command to check if the player can use it.
+     * @param colonyId the id of the colony.
+     * @return boolean.
+     */
+
+    public boolean canPlayerUseCommand(final EntityPlayer player, final Commands theCommand, final int colonyId)
+    {
+        if (isPlayerOpped(player, theCommand.toString()))
+        {
+            return true;
+        }
+
+        final Colony chkColony = ColonyManager.getColony(colonyId);
+        if(chkColony == null)
+        {
+            return false;
+        }
+        return canCommandSenderUseCommand(theCommand)
+                && canRankUseCommand(chkColony, player);
+    }
+
+    /**
+     * Will check to see if play is Opped for the given command name.
+     *
+     * @param sender  to check the player using the command.
+     * @param cmdName the name of the command to be checked.
+     * @return boolean
+     */
+    @NotNull
+    public boolean isPlayerOpped(@NotNull final ICommandSender sender, String cmdName)
+    {
+        if (sender instanceof EntityPlayer)
+        {
+            return FMLCommonHandler.instance().getMinecraftServerInstance().getPlayerList()
+                     .canSendCommands(((EntityPlayer) sender).getGameProfile());
+        }
+        return true;
+    }
+
+    /**
+     * Will check the config file to see if players are allowed to use the command that is sent here.
+     *
+     * @param theCommand which command to check if the player can use it
+     * @return boolean
+     */
+    public boolean canCommandSenderUseCommand(Commands theCommand)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 13, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)